### PR TITLE
Skip indexing for some files

### LINKED ISSUE #2106: Seeing unexpected values for "type" in file browser
(Started in #2089, but a new bug to reflect new, limited scope.)

Other facets seem fine, but the type facet returns stray zero values (with `mincount` removed). I had expected the list for every dataset to be the same, so I am probably not thinking about it right.

[Transcription factor ChIP-seq in expanded human hematopoietic stem and progenitor cells](https://test.stemcellcommons.org/data_sets/be9af94c-43fb-44a0-b357-12fdf0d8d008/#/files/):

> Raw Data File 10
> Assay Name 0
> Extract Name 0

[Molecular Analysis of antigen-specific B cell responses](https://test.stemcellcommons.org/data_sets/6a497724-acce-4e0d-8907-b98f0542061d/#/files/)

> Array Data File 14
>Derived Array Data Matrix File 4
> Derived Array Data File 2
> Data Transformation Name 0
> Extract Name 0
> Hybridization Assay Name 0
> Labeled Extract Name 0
> Scan Name 0

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #2106? Does this seem like a reasonable approach? I still want to:
- Simplify the solr queries we construct, since these documents won't even be in the index now. (ie, add #2288 back in.)
- Move this list to configuration.
- Review other lists of hard-coded document types.

### CODE REVIEW

**Comment by mccalluc** [comment]
On second thought, perhaps not a config, but class constant. I think this is all that is needed now?

File: refinery/data_set_manager/search_indexes.py [hunk 1]
```diff
@@ -71,6 +72,8 @@ def _assay_data(self, object):
     # https://groups.google.com/forum/?fromgroups#!topic/django-haystack/g39QjTkN-Yg
     # http://stackoverflow.com/questions/7399871/django-haystack-sort-results-by-title
     def prepare(self, object):
+        if object.type not in Node.INDEXED_FILES:
+            raise SkipDocument()
```
Comment (scottx611x) [inline refinery/data_set_manager/search_indexes.py]: Can `SkipDocument` accept a message? Could be useful for debugging.

**Comment by mccalluc** [comment]
@ngehlenborg : Can you schedule time to give input on this, or delegate someone who can make a call? There are several fairly simple changes we could make that would fix the behavior, but we're not sure which is best.

**Comment by mccalluc** [comment]
@scottx611x : Talked with Nils: Coming out of that, the idea was to show all documents, but for all the doc types we don't recognize, change the type, *but also* add a new field to the object so we could theoretically export the original.

With a little distance, that feels too complicated, and too fragile. How would you feel about merging this original PR, and putting the rest on the back burner?

**Comment by scottx611x** [comment]
@mccalluc I agree on the fragility of: `for all the doc types we don't recognize, change the type, but also add a new field to the object so we could theoretically export the original`

Let me look over this PR again first and I'll give a better opinion

**Comment by scottx611x** [comment]
@mccalluc I'd be okay with merging this as long as we move our future intentions into a new issue

**Pull request merged**